src/main/java/pl/touk/repository/InMemoryParkingRateRepository.java

```diff
@@ -0,0 +1,54 @@
+package pl.touk.repository;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Repository;
+import pl.touk.model.ParkingRate;
+import pl.touk.model.VipCustomer;
+
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.stream.Collectors;
+
+@Slf4j
+@Repository
+public class InMemoryParkingRateRepository implements ParkingRateRepository {
+
+    private AtomicLong idGenerator = new AtomicLong();
+    private Map<Long, ParkingRate> rates = new ConcurrentHashMap<>();
+    private Map<Long, VipCustomer> vips = new ConcurrentHashMap<>();
+
+    @Override
+    public void save(List<ParkingRate> parkingRates) {
+        log.debug("Save " + parkingRates.toString());
+        parkingRates.forEach(parkingRate -> {
+            rates.put(idGenerator.getAndIncrement(), parkingRate);
+            parkingRate.setId(idGenerator.get());
```

Comment:
nie rozjadą Ci się tu idiki? tak na oko to w mapie ustawiasz wartość `id` a przez `setId ` dajesz `id + 1`

Reply:
Owszem rozjadą, zapomniałem tutaj poprawić powinno być: parkingRate.setId(idGenerator.getAndDecrement());